src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -143,6 +129,41 @@ public function __construct($mongo, array $options = [], float $initialTtl = 300
         }
     }
 
+    /**
+     * Extract default database and collection from given connection uri and remove collection querystring.
+     */
+    private function skimUri(string $uri): string
```

Comment:
I'm not comfortable with this method, a lot of complex logic to extract parameter from URI mixing parse_str and strpos.

Givent the format of the initial `DSN` is defined by symfony, wouldn't it be safer to build the mongo URI instead?

Reply:
the reason i chose to use strpos to re-assemble was to simplify the re-assembly. there is no reverse of parse_str in php. see https://www.php.net/manual/en/function.parse-url.php#106731

I'm trying not to get too involved with touching the URI. What would you suggest that is simpler?